### PR TITLE
Datatype id use schema

### PR DESCRIPTION
**Pull request by daco101**
This PR changes the models to be more in line with the qdm spec - it adds nested qdm ids to data elements.

Pull requests into cqm-models require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/BONNIE-1666
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] If there were any JavaScript changes, this PR has updated the `dist` directory using `npm run dist`
- [x] If applicable, the library version number in `package.json` and `cqm-models.gemspec` has been updated. NA
- [x] All changes can be reproduced by running the generator script

**Bonnie Reviewer:**

Name: Group review
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code
- [x] You have executed the generator script, and made sure no changes were made that the generator did not reproduce itself

**Cypress Reviewer:**

Name: David Czulada
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code
- [x] You have executed the generator script, and made sure no changes were made that the generator did not reproduce itself

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,4 +2,8 @@ source 'https://rubygems.org'
 
 git_source(:github) { |repo_name| "https://github.com/#{repo_name}" }
 
+group :test, :development do
+  gem 'pry-byebug'
```
Comment (holmesie) [inline Gemfile]: This does not belong here. If you want to add dependencies like this to a Gem, even if for development purposes, do it in the gemspec.

Also, byebug is already a development dependency. Why this extra dependency? If you want to step through things, why not just use byebug in the first place?

File: lib/generate_models.rb [hunk 2]
```diff
@@ -162,19 +160,19 @@
...
-    patient_template = File.read('templates/patient_template.js.erb')
-    patient_renderer = ERB.new(patient_template, nil, '-')
-    attrs_with_extras = attributes + extra_fields_js
-    puts '  ' + file_path + datatype + '.js'
-    File.open(file_path + datatype + '.js', 'w') { |file| file.puts patient_renderer.result(binding) }
-  else
-    attrs_with_extras = attributes + extra_fields_js
-    puts '  ' + file_path + datatype + '.js'
-    File.open(file_path + datatype + '.js', 'w') { |file| file.puts renderer.result(binding) }
+  renderer = default_renderer
```
Comment (holmesie) [inline lib/generate_models.rb]: This block change seems... awkward. Why not just add another `elsif datatype == 'Id'`?

File: app/models/qdm/basetypes/data_element.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
+require_relative('../id.rb')
```
Comment (dczulada) [inline app/models/qdm/basetypes/data_element.rb]: In models.rb the qdm/id needs to be moved before require_relative 'qdm/basetypes/data_element'

for example, (app/models/models.rb) could be modified to look something like this

require 'mongoid'

# Generated models with base type dependencies
require_relative 'qdm/id'

# base types
require_relative 'qdm/basetypes/code'
require_relative 'qdm/basetypes/interval'
require_relative 'qdm/basetypes/quantity'
require_relative 'qdm/basetypes/data_element'

**Comment by mayerm94** [comment]
Looks good

**Pull request merged**